### PR TITLE
NestedFolders: Permission for creating and editing

### LINKED ISSUE #67392: NestedFolders: Checkboxes and actions permissions
Get user permissions and decide whether or not to show:
- Checkboxes
- Move/Delete
- Create new thing

Permissions aren't as straightforward any more, so take inspiration from how they appear in `ManageDashboardsNew.tsx`

### PR DESCRIPTION
**Pull request by tskarhed**
**What is this feature?**
Adds permissions to nested folders. Things are displayed based on your permissions.

**Why do we need this feature?**
Otherwise chaos ensues.

**Who is this feature for?**

Everybody who likes to 
**Which issue(s) does this PR fix?**:

Fixes #67392

**Special notes for your reviewer:**

Try this with RBAC permissions in Grafana Enterprise

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/browse-dashboards/components/DashboardsTree.tsx [hunk 1]
```diff
@@ -95,9 +99,12 @@ export function DashboardsTree({
       Header: 'Tags',
       Cell: TagsCell,
     };
-
-    return [checkboxColumn, nameColumn, typeColumn, tagsColumns];
-  }, [onItemSelectionChange, onAllSelectionChange, onFolderClick]);
+    const columns = [nameColumn, typeColumn, tagsColumns];
+    if (checkboxColumn) {
+      columns.unshift(checkboxColumn);
+    }
```
[the remark this answers is not in the document]
Comment (joshhunt) [reply public/app/features/browse-dashboards/components/DashboardsTree.tsx:105]: Something like

```tsx
import { isTruthy } from '@grafana/data';

// ...

const columns = [
  showCheckboxes && checkboxColumn,
  nameColumn,
  typeColumn,
  tagsColumn
].filter(isTruthy)
```

perhaps?

File: public/app/features/browse-dashboards/BrowseDashboardsPage.tsx [hunk 2]
```diff
@@ -68,9 +83,15 @@ const BrowseDashboardsPage = memo(({ match }: Props) => {
           <AutoSizer>
             {({ width, height }) =>
               isSearching ? (
-                <SearchView key={rerender} width={width} height={height} />
+                <SearchView key={rerender} showCheckboxes={canEditInFolder} width={width} height={height} />
```
Comment (joshhunt) [inline public/app/features/browse-dashboards/BrowseDashboardsPage.tsx:86]: lets rename this to `canManage` or `canSelect` something. We want to describe capabilities, not figure implementation details.

File: public/app/features/browse-dashboards/components/DashboardsTree.test.tsx [hunk 3]
```diff
@@ -42,11 +45,31 @@ describe('browse-dashboards DashboardsTree', () => {
     expect(screen.queryByText(dashboard.item.title)).toBeInTheDocument();
     expect(screen.queryByText('Dashboard')).toBeInTheDocument();
     expect(screen.queryByText(assertIsDefined(dashboard.item.tags)[0])).toBeInTheDocument();
+    expect(screen.getByTestId(selectors.pages.BrowseDashbards.table.checkbox(dashboard.item.uid))).toBeInTheDocument();
+  });
+
+  it('does not render checkbox when disabled', () => {
```
Comment (joshhunt) [inline public/app/features/browse-dashboards/components/DashboardsTree.test.tsx:51]: 👍🏻

**Review by joshhunt — APPROVED** [review]
Looks great. One minor suggestion about renaming a prop, but otherwise all good

File: public/app/features/browse-dashboards/BrowseDashboardsPage.tsx (hunk 2 above)
Comment (tskarhed) [reply public/app/features/browse-dashboards/BrowseDashboardsPage.tsx:86]: @joshhunt Renaming `showCheckboxes` I assume?

Comment (joshhunt) [reply public/app/features/browse-dashboards/BrowseDashboardsPage.tsx:86]: Sorry, yes 👍🏻

**Review by joshhunt — APPROVED** [review]

**Pull request merged**